framework/include/utils/MooseEnumBase.h

```diff
@@ -109,11 +95,39 @@ class MooseEnumBase
   std::set<MooseEnumItem>::const_iterator find(int id) const;
   ///@}
 
+protected:
+  MooseEnumBase();
+
+  ///@{
+  /**
+   * Methods to add possible enumeration value to the enum.
+   *
+   * The MooseEnum/MultiMooseEnum are not designed to be modified, with respect to the list
+   * of possible values. However, this is not the case for the ExecFlagEnum which is a special
+   * type of MultiMooseEnum for managing the "execute_on" flags. These methods are used by
+   * ExecFlagEnum to allow users to modify the available execute flags for their object.
+   */
+  void addEnumerationNames(const std::string & names);
+  void addEnumerationName(const std::string & raw_name);
+  void addEnumerationName(const std::string & name, const int & value);
+  void addEnumerationName(const MooseEnumItem & item);
```

Comment:
These should be protected in the base and are basically just a replacement for the filenames method that was there, using several methods so that extra flexibility is available down the line.